src/ecalc/libraries/libecalc/common/libecalc/core/consumers/consumer_system.py

```diff
@@ -279,8 +279,31 @@ def _topologically_sort_consumers_by_crossover(crossover: List[int], consumers:
     def _get_crossover_rates(
         max_rate: List[float], rates: List[List[float]]
     ) -> Tuple[NDArray[np.float64], List[List[float]]]:
+        """
+        This function is run over a single consumer only, and is normally run in a for loop
+        across all "dependent" consumers in the consumer system, such as here, in a consumer system.
+
+        This function will calculate:
+         1. the additional crossover rate that is required if the total incoming
+        rate exceeds the max rate capacity for the consumer in question.
+         2. the rates within capacity of the given consumer
```

Comment:
Nice explanation, could this be under Returns or is that unclear when using a tuple return?

Reply:
Forgot about returns, yes, the idea was that it should be there :p